packages/shared/src/components/search/SearchBarSuggestion.tsx

```diff
@@ -27,33 +28,33 @@ export const SearchBarSuggestion = ({
   const { trackEvent } = useContext(AnalyticsContext);
 
   useEffect(() => {
-    if (!props.isHistory && props.suggestion?.id) {
+    if (!props.isHistory && props?.id) {
       trackEvent({
         event_name: AnalyticsEvent.Impression,
         target_type: TargetType.SearchRecommendation,
-        target_id: props.suggestion.id,
-        feed_item_title: props.suggestion.prompt,
+        target_id: props.id,
+        feed_item_title: props.prompt,
         extra: JSON.stringify({ origin }),
       });
     }
 
     // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [props.suggestion?.id]);
+  }, [props.id]);
```

Comment:
It should never refire the tracking though, so with this in mind we need a simple state to keep track of whether it's been fired on this render or not.
It can be as simple as a useState boolean that get's set inside the evalaution.

I know it's probably overkill and never triggers, but let's be safe if something changes down the line.

Reply:
ref is better to not trigger re-renders, but yeah, if we want to make sure this only fires on component mount then this is legit way.

Reply:
Updated based on your feedback, we have a bool internal state which should make extra sure these are not emitted multiple times